### ISSUE TITLE
Suggestion to replace non-React tooltip with pods-tooltip web components

### ISSUE THREAD

**Issue by heybran** [pods-framework/pods#7123]
### Problem to Solve

Make non-React tooltip styles behave consistently.

### Proposed Solution

As reported on #7118, there's a few tooltips (non-React tooltip) that has style/accessibility issue, it would be nice to replace those tooltips with a standalone [`pods-tooltip`](https://github.com/heybran/pods-tooltip) custom element so tooltip styles can be encapsulated inside shadowRoot. Another cool thing about web components is that we can safely update styles without worrying it breaking other elements.

I have coded out a first version of `pods-tooltip`, it's looking pretty good so far? 

Still few improvements needed, hopefully I can get it done soon.

1. Best not to overlap with main sidebar
2. If anchor link inside tooltip, they should be accessed by keyboard
3. Same as react tooltip, make it visible and close when click outside (a bug at the moment)

<img width="1162" alt="Screenshot 2023-07-26 at 21 07 24" src="https://github.com/pods-framework/pods/assets/75633537/01ca0f5c-9a8d-4fb7-a1f6-4c66e3a3516e">

**Comment by heybran** [comment]
## Notes


### TODO

- Added `pods-tooltip.js` file https://github.com/heybran/pods/commit/0a88aabc0f921ce859f963cca12a7f2950f13bbe , although codes format doesn't match with Wordpress coding styles, need to find a way to auto format this file.
- Need to minimize the `pods-tooltip.js` file, forgot about that.

### Updates

1. Replace old `png` help icon (non-React ones) with a svg icon that was included in `pods-tooltip` component 

    <img width="989" alt="Screenshot of pods-tooltip when focused" src="https://github.com/pods-framework/pods/assets/75633537/620ee357-ee69-4873-a90e-199a5d6ba3f4">

2. Added keyboard support for tooltip itself and any anchor inside, if any

    <img width="989" alt="Screenshot of pods-tooltip when focused" src="https://github.com/pods-framework/pods/assets/75633537/9ecf2ca1-2124-4fd5-bd06-f81850147f7f">



### Questions

1. I enqueue script of `pods-tooltip.js` on Line935 & 936 https://github.com/pods-framework/pods/blob/fb70bcc5f44a4003e297a1e49541f8bb439854a9/includes/general.php#L935 https://github.com/pods-framework/pods/blob/fb70bcc5f44a4003e297a1e49541f8bb439854a9/includes/general.php#L936 and render that custom element to the DOM on Line 972 https://github.com/pods-framework/pods/blob/fb70bcc5f44a4003e297a1e49541f8bb439854a9/includes/general.php#L972
   Since `pods_help()` is the function responsible for `non-React` tooltip, and if this solution is accepted, does it mean we can clean up this `pods_help()`, meaning removing those `jQuery` related tooltip codes?

**Comment by heybran** [comment]
Heya @sc0ttkclark 

If you see this and have some time, I have a question.

I enqueue script of `pods-tooltip.js` on Line935 & 936 https://github.com/pods-framework/pods/blob/fb70bcc5f44a4003e297a1e49541f8bb439854a9/includes/general.php#L935 https://github.com/pods-framework/pods/blob/fb70bcc5f44a4003e297a1e49541f8bb439854a9/includes/general.php#L936 and render that custom element to the DOM on Line 972 https://github.com/pods-framework/pods/blob/fb70bcc5f44a4003e297a1e49541f8bb439854a9/includes/general.php#L972
   Since `pods_help()` is the function responsible for `non-React` tooltip, and if this solution is accepted, does it mean we can clean up this `pods_help()`, meaning removing those `jQuery` related tooltip codes?

After this, I get all I need to create a PR for this issue.

**Comment by sc0ttkclark** [comment]
Yes, we'd need to update `pods_help()` usage but also what we use in React would be at `ui/js/dfv/src/components/help-tooltip.js`

Whatever solution we go with here, we should ensure it is accessible and follows the necessary standards for screen readers as well.

Using one tooltip solution for both would be great. I'm not 100% sold on the `<pods-tooltip>` custom HTML tag and how that affects accessibility.

**Comment by heybran** [comment]
I was a Contributor Day Accessibility table yesterday at WordCamp Asia and learn about this trac ticket: https://core.trac.wordpress.org/ticket/51006

I'm assuming it would be best to handle the tooltip for Pods using core API once that trac ticket has been implemented into core?
